### PR TITLE
Include buildx (and use a build argument so as not to edit the dockerfile)

### PR DESCRIPTION
**Pull request by ciaranmcnulty**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -8,7 +8,7 @@
 # Building & tagging
 
  * `export TAG_VERSION=19.03.13` (update this)
- * `docker build -t asgrim/docker-with-compose:$TAG_VERSION .`
+ * `docker build --build-arg TAG_VERSION=$TAG_VERSION -t asgrim/docker-with-compose:$TAG_VERSION .`
  * `docker push asgrim/docker-with-compose:$TAG_VERSION`
  * `git tag -s $TAG_VERSION`
```
Comment (ciaranmcnulty) [inline README.md]: Not sure this is needed now?

Comment (asgrim) [reply README.md]: Yeah probably not :)

File: Dockerfile [hunk 2]
```diff
@@ -1,6 +1,11 @@
+ARG TAG_VERSION=19.03.13
+ARG ALPINE_VERSION=3.13
+
+FROM docker:$TAG_VERSION as docker-base
+
 FROM alpine:3.13
```
Comment (asgrim) [inline Dockerfile]: Should this be

```suggestion
FROM alpine:$ALPINE_VERSION
```

?

Comment (ciaranmcnulty) [reply Dockerfile]: pushed too soon! other errors too

File: Dockerfile [hunk 3]
```diff
@@ -1,6 +1,13 @@
-FROM alpine:3.13
+ARG TAG_VERSION=19.03.13
+ARG ALPINE_VERSION=3.13
 
-COPY --from=docker:19.03.13 /usr/local/bin/docker /usr/local/bin
+FROM docker:$TAG_VERSION as docker-base
+
+FROM alpine:ALPINE_VERSION
+
+ENV CRYPTOGRAPHY_DONT_BUILD_RUST=1
```
Comment (ciaranmcnulty) [inline Dockerfile]: This lets it build on my machine 🤷

File: Dockerfile [hunk 4]
```diff
@@ -1,6 +1,13 @@
...
 
 RUN apk add --no-cache py3-pip python3-dev libffi-dev openssl-dev curl gcc libc-dev make && \
     pip3 install docker-compose
+
+COPY --from=docker-base /usr/local/bin/docker /usr/local/bin
```
Comment (asgrim) [inline Dockerfile:14]: What is the difference in doing `FROM docker:$TAG_VERSION as docker-base` and using it in the `COPY` here, versus doing just:

```
COPY --from=docker:$TAG_VERSION /usr/local/bin/docker /usr/local/bin
```

(maybe that's not possible? just eyeballing it)

Comment (ciaranmcnulty) [reply Dockerfile:14]: Docker won't expand it inside a `--from` (which is probably a bug/oversight)

**Comment by ciaranmcnulty** [comment]
please don't merge yet, I'm a dummy

**Comment by ciaranmcnulty** [comment]
Should be good to review now

**Comment by asgrim** [comment]
Thganks :+1:

**Pull request merged**